mllib/src/main/scala/org/apache/spark/ml/tree/impl/DecisionTreeMetadata.scala

```diff
@@ -115,7 +122,10 @@ private[spark] object DecisionTreeMetadata extends Logging {
     }
     require(numFeatures > 0, s"DecisionTree requires number of features > 0, " +
       s"but was given an empty features vector")
-    val numExamples = input.count()
+    val (numExamples, weightSum) = input.aggregate((0L, 0.0))(
```

Comment:
although by idea none of these changes should impact performance much, have you had any chance to verify the execution time is the same as before?